client/db.go

```diff
@@ -161,6 +161,22 @@ type Store interface {
 	// this [Store].
 	GetAllCollections(context.Context) ([]Collection, error)
 
+	// GetSchemaByName returns the all schema versions with the given name.
+	GetSchemaByName(context.Context, string) ([]SchemaDescription, error)
+
+	// GetSchemaByVersionID returns the schema description for the schema version of the
+	// ID provided.
+	//
+	// Will return an error if it is not found.
+	GetSchemaByVersionID(context.Context, string) (SchemaDescription, error)
+
+	// GetSchemaByRoot returns the all schema versions for the given root.
+	GetSchemaByRoot(context.Context, string) ([]SchemaDescription, error)
+
+	// GetAllSchema returns all schema versions that currently exist within
+	// this [Store].
+	GetAllSchema(context.Context) ([]SchemaDescription, error)
```

Comment:
These are great additions to the API! It really helps clarify the workings of schemas and collections.

Reply:
hey @nasdf I just wanted to point out that we try to stick to the conventional commenting style https://conventionalcomments.org/